articles/kotlin-extensions/index.md

```diff
@@ -0,0 +1,180 @@
+
+```Kotlin
+fun ImageView.loadUrl(url: String){
+    Glide.with(this.context).load(url).into(this)
+}
+
+//In your activity
+imageView.loadUrl(url)
+
+```
+
+From this, we can get the `context` and `target`(view) from the extended class. This code is much cleaner and easier to read than the object. You can also pass in the `int` id's of resources from your application.
+
+```Kotlin
+fun ImageView.loadDrawable(id: Int) {
+    Glide.with(this.context).asDrawable().load(id).into(this)
+}
+
+//In your activity
+imageView.loadDrawable(R.drawable.ic_android)
+```
+
+The xml for the two images :
+
+```xml
+<LinearLayout
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:padding="10dp">
+
+    <androidx.appcompat.widget.AppCompatImageView
+        android:id="@+id/url_image"
+        android:layout_width="0dp"
+        android:layout_height="100dp"
+        android:layout_margin="10dp"
+        android:layout_weight="1" />
+
+    <androidx.appcompat.widget.AppCompatImageView
+        android:id="@+id/drawable_image"
+        android:layout_width="0dp"
+        android:layout_height="100dp"
+        android:layout_margin="10dp"
+        android:layout_weight="1" />
+
+</LinearLayout>
+```
+
+With that, you get only one line of code that can be reused. This also improves performance by creating only one function to do all the work. We can also extend other view classes like `TextView`. We can use it to make changes to a specific TextView.
+
+```Kotlin
+private val chars = ('a'..'Z') + ('A'..'Z') + ('0'..'9')
+private fun random() = List(5) { chars.random() }.joinToString("")
+
+fun TextView.randomText(){
+    this.text = random()
+}
+```
+
+In the code above, we first created a string of characters from a-z in both cases. Then we also concatenate the numbers from 0 to 9. This is done using the range operator in Kotlin. You can read more about the operator in the [official documentation](https://kotlinlang.org/docs/reference/ranges.html). This reduces the lines of code we would have written if we used the loop structures.
+
+Then we create a list with 5 items. Then get random characters from the string. Use the `joinToString` function to create a new string of 5 characters. None of them are meant to have any semantic meaning. We are using it to populate the TextView. The extension function `randomText` adds the random string to our TextView.
+
+```xml
+<!-- activity_main.xml -->
+
+<com.google.android.material.textview.MaterialTextView
+    android:id="@+id/random"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_gravity="center" />
+```
+
+```kotlin
+random.randomText()
+```
+
+We can also use it to make it easier to perform specific actions in activities and fragments. Any component that has access to the context can show `Toast`. `Snackbars` are also a common feature. To create the extension classes for them:
```

Comment:
🚀  Hi @LinusMuema  - doing a preliminary review on the Kotlin article - it seems very informative.
This sentence confused me a bit - (it may have been due to not having a background with Android or Kotlin) but the toast and snackbars.
"`Snackbars` are also a common feature." for... extending a class? 
I think the sentence could use a bit more context to help avoid any confusion - also the following line 
"To create the extension classes for them:" Im assuming the classes are the examples below?

Reply:
Hi @hectorkambow 
Sorry for the late reply. To remove any confusion among readers, I think their purpose is only one: 'To display messages and short pieces of information'. And yes, the functions below are for them